sdk/src/main/scala/com/horizen/api/http/SidechainBlockActor.scala

```diff
@@ -88,47 +90,31 @@ class SidechainBlockActor[PMOD <: PersistentNodeViewModifier, SI <: SidechainSyn
 
   }
 
-  // Note: It should be used only in regtest
-  protected def generateSidechainBlocks: Receive = {
-    case GenerateSidechainBlocks(blockCount) =>
-      // Try to forge blockCount blocks, collect their ids and wait for
-      var generatedIds: Seq[ModifierId] = Seq()
-      for (i <- 1 to blockCount) {
-        val future = forgerRef ? TryForgeNextBlock
-        Await.result(future, timeoutDuration).asInstanceOf[Try[ModifierId]] match {
-          case Success(id) =>
-            generatedIds = id +: generatedIds
-            if (i == blockCount) {
-              // Create a promise, that will wait for blocks applying result from Node
-              val prom = Promise[Try[Seq[ModifierId]]]()
-              generatedBlockGroups += (id -> generatedIds)
-              generatedBlocksPromises += (id -> prom)
-              sender() ! prom.future
-            }
-
-          case Failure(ex) =>
-            sender() ! Future[Try[Seq[ModifierId]]](Failure(ex))
-        }
-      }
-  }
-
-  protected def tryToSubmitBlock: Receive = {
-    case SubmitSidechainBlock(blockBytes: Array[Byte]) =>
-      val future = forgerRef ? TrySubmitBlock(blockBytes)
-      Await.result(future, timeoutDuration).asInstanceOf[Try[ModifierId]] match {
-        case Success(id) =>
+  protected def processTryForgeNextBlockForEpochAndSlotMessage: Receive = {
+    case messageToForger @ TryForgeNextBlockForEpochAndSlot(epochNumber, slotNumber) =>
+    {
+      val future = forgerRef ? messageToForger
+      val result = Await.result(future, timeoutDuration).asInstanceOf[ForgeResult]
+      result match {
+        case ForgeSuccess(block) => {
           // Create a promise, that will wait for block applying result from Node
           val prom = Promise[Try[ModifierId]]()
-          submitBlockPromises += (id -> prom)
+          submitBlockPromises += (block.id -> prom)
+
+          //and only then apply new block
+          sidechainNodeViewHolderRef ! LocallyGeneratedModifier[SidechainBlock](block)
+
           sender() ! prom.future
+        }
+        case SkipSlot => sender() ! Future(Failure(new SkipSlotExceptionMessage()))
 
-        case Failure(ex) =>
-          sender() ! Future[Try[ModifierId]](Failure(ex))
+        case ForgeFailed(exception) => sender() ! Future(exception)
```

Comment:
I still don't get why need to return `ForgeResult`, that separates `SkipSlot` and `ForgeFailed`.
It should return Try[SidechainBlock]. So in a case, when forger is not eligible to forge for required slot, it will return an exception with proper message to `SidechainBlockActor`. 
But instead now `SidechainBlockActor` as any other receiver must somehow to wrap this case.
It can be a solution, if special behavior needed from `SidechainBlockActor` when exactly `SkipSlot` received...